fedimint-core/src/query.rs

```diff
@@ -80,16 +79,19 @@ pub struct UnionResponses<R> {
     responses: HashSet<PeerId>,
     existing_results: Vec<R>,
     current: CurrentConsensus<Vec<R>>,
-    required: usize,
+    threshold: usize,
 }
 
 impl<R> UnionResponses<R> {
-    pub fn new(required: usize) -> Self {
+    pub fn new(total_peers: usize) -> Self {
+        let max_evil = (total_peers - 1) / 3;
+        let threshold = total_peers - max_evil;
```

Comment:
So if I understand correctly, the reason we're getting rid of `EventuallyConsistent` and not using `one_honest` anymore is because `one_honest` is not safe and can lead to race conditions. Therefore, we actually always need to wait for `total - max_evil` to truly know if a transaction has been included in consensus or not. Correct?

Reply:
Yes, using one_honest instead of threshold leads to a race condition. Furthermore, the old CurrentConsensus did not implement retries if no quorum was reached which lead to a different race condition, hence we always use retries now just as EventuallyConsistent used to which is why we removed EventuallyConsistent as retries were the only difference to CurrentConsensus.